riscv-c-api.md

```diff
@@ -621,3 +621,156 @@ statements, including both RISC-V specific and common operand modifiers.
 | ------------ | --------------------------------------------------------------------------------- | ----------- |
 | z            | Print `zero` (`x0`) register for immediate 0, typically used with constraints `J` |             |
 | i            | Print `i` if corresponding operand is immediate.                                  |             |
+
+## Function Multi-version
+
+Function multi-versioning (FMV) allows selecting the appropriate function based on the runtime environment. The binary may contain multiple versions of the function, with the compiler generating all supported versions and selecting the appropriate one during the runtime.
+
+This feature is activated by the `target_version/target_clones` function attributes.
+
+### Extension Bitmask
+
+When generating the resolver function for FMV, a method is required to retrieve environment information. This is achieved through two bitmask structures: `__riscv_feature_bits` for standard extensions and `__riscv_vendor_feature_bits` for vendor-specific extensions.
+
+The bitmask structures use the following definitions:
+
+```c
+struct {
+    unsigned length;
+    unsigned long long features[];
+} __riscv_feature_bits;
+
+struct {
+    unsigned vendorID
+    unsigned length;
+    unsigned long long features[];
+} __riscv_vender_feature_bits;
+```
+
+- `length`: Represents the number of elements in the features array.
+- `features`: An `unsigned long long` array where each bit indicates the 1 for enable or 0 for disable of a corresponding extension.
```

Comment:
I think the language here needs to be tightened up - `1` means "you definitely can use this extension", but `0` needs to mean "software must not assume this extension is implemented", given what happened with the `b` extension being ratified as a collection of `Zb*` extensions after-the-fact (similar information is defined in the B specification). This also covers if an OS wants to turns off a feature for any reason, which it could do by collaborating with the C library.